wolfcrypt/src/pwdbased.c

```diff
@@ -198,7 +198,7 @@ int wc_PBKDF2(byte* output, const byte* passwd, int pLen, const byte* salt,
     if (buffer == NULL)
         return MEMORY_E;
     hmac = (Hmac*)XMALLOC(sizeof(Hmac), NULL, DYNAMIC_TYPE_HMAC);
-    if (buffer == NULL)
+    if (hmac == NULL)
```

Comment:
Please also handle the `buffer` free case here. Otherwise `buffer` will leak.
`XFREE(buffer, NULL, DYNAMIC_TYPE_TMP_BUFFER);`

Reply:
Thanks David